heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -295,6 +295,22 @@
    */
   public static final String TOPOLOGY_COMPONENT_OUTPUT_BPS = "topology.component.output.bps";
 
+  /**
+   * Default number of cpu cores per component to be reserved for this topology
+   */
+  public static final String TOPOLOGY_COMPONENT_DEFAULT_CPU = "topology.component.default.cpu";
+  /**
+   * Default amount of ram per component to be reserved for this topology.
+   * In bytes.
+   */
+  public static final String TOPOLOGY_COMPONENT_DEFAULT_RAM = "topology.component.default.ram";
```

Comment:
Is this statement accurate: all instances of a component together will get this amount of RAM, be default? For e.g. if a comp C has instance i1 and i2 and default RAM is 2 GB, i1 will be assigned 1GB of ram.

On the other hand, if this is instance default, how is different from current instance ram config?

Reply:
@kramasamy @nwangtw 
I am a bit confused. 
1. The config name is `COMPONENT_DEFAULT`, but it is use to configure `INSTANCE_DEFAULT`. Should this be renamed? 
1. Why are we replacing existing configs for cpu and disk, `heron.resources.instance.cpu` and `heron.resources.instance.disk`

Reply:
1. It is a good question. I think both makes sense, in a different way. If you consider the next step, per component resource config, component_default makes more sense. If you consider that all instances are using the config as default value, instance_default makes more sense. I am thinking instance_default might feel better now. Open to other thoughts.

2. replacing? existing configs are used as fallback values, unless there is a bug.